src/app/app/secretaria/oficios/_components/OficioForm.tsx

```
@@ -51,20 +50,26 @@ function textToParagraphHtml(text: string) {
+  const [aiError, setAiError]           = useState<string | null>(null);
+  const [isAiPending, startAiTransition] = useTransition();
+  const [isSubmitPending, startSubmitTransition] = useTransition();
 
   useEffect(() => {
     if (!isAiModalOpen) return;
-    function handleKeyDown(e: KeyboardEvent) {
-      if (e.key === 'Escape') setIsAiModalOpen(false);
-    }
-    document.addEventListener('keydown', handleKeyDown);
-    return () => document.removeEventListener('keydown', handleKeyDown);
+    const onKey = (e: KeyboardEvent) => { if (e.key === 'Escape') closeAiModal(); };
```

ESLint

[error] Error: Cannot access variable before it is declared

`closeAiModal` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/src/app/app/secretaria/oficios/_components/OficioForm.tsx:62:67
  60 |   useEffect(() => {
  61 |     if (!isAiModalOpen) return;
> 62 |     const onKey = (e: KeyboardEvent) => { if (e.key === 'Escape') closeAiModal(); };
     |                                                                   ^^^^^^^^^^^^ `closeAiModal` accessed before it is declared
  63 |     document.addEventListener('keydown', onKey);
  64 |     return () => document.removeEventListener('keydown', onKey);
  65 |   }, [isAiModalOpen]);

/home/jailuser/git/src/app/app/secretaria/oficios/_components/OficioForm.tsx:67:3
  65 |   }, [isAiModalOpen]);
  66 |
> 67 |   const closeAiModal = () => {
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 68 |     setIsAiModalOpen(false);
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 69 |     setAiInstruction('');
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 70 |     setAiError(null);
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 71 |   };
     | ^^^^^ `closeAiModal` is declared here
  72 |
  73 |   const {
  74 |     register,

(react-hooks/immutability)